R/bundlePackage.R

```diff
@@ -24,136 +25,68 @@ bundlePackages <- function(appDir,
     documentsHavePython = documentsHavePython
   )
   deps <- snapshotRDependencies(appDir, inferredRDependencies, verbose = verbose)
+  checkBundlePackages(deps, call = error_call)
 
-  packageMessages <- c()
-  errorMessages <- c()
-  packages <- list()
-
-  # construct package list from dependencies
-  for (i in seq_len(nrow(deps))) {
-    name <- deps[i, "Package"]
-
-    # get package info
-    info <- as.list(deps[i, c("Source", "Repository")])
-
-    # include github package info
-    info <- c(info, as.list(deps[i, grep("Github", colnames(deps), perl = TRUE, value = TRUE)]))
-
-    # get package description; note that we need to remove the
-    # packageDescription S3 class from the object or jsonlite will refuse to
-    # serialize it when building the manifest JSON
+  deps$description <- lapply(deps$Package, function(nm) {
+    # Remove packageDescription S3 class so jsonlite can serialize
     # TODO: should we get description from packrat/desc folder?
-    info$description <- suppressWarnings(unclass(utils::packageDescription(name)))
-
-    # TODO(HW): should this code be removed? I don't see how it is ever reached
-    # if description is NA, application dependency may not be installed
-    if (is.na(info$description[1])) {
-      errorMessages <- c(
-        errorMessages,
-        paste0(
-          capitalize(assetTypeName), " depends on package \"", name, "\"",
-          "but it is not installed. Please resolve before continuing."
-        )
-      )
-      next
-    }
-
-    # validate package source (returns a message if there is a problem)
-    packageMessages <- c(packageMessages, validatePackageSource(deps[i, ]))
+    unclass(utils::packageDescription(nm))
+  })
 
-    # good to go
-    packages[[name]] <- info
-  }
+  # Connect will generally use the packrat/packrat.lock file; the data
+  # in the manifest is used primarily by shinyapps.io, which only supports
+  # Github remotes, not Bitbucket or Gitlab.
```

Comment:
shinyapps.io wants Github information in the outer package record and does not consider the description sub-record. Connect generally prefers the packrat.lock, but when it only has a manifest, it uses the GitHub/Remotes fields from the description sub-record.